src/features/antora/antoraSupport.ts

```diff
@@ -0,0 +1,86 @@
+import vscode, { CancellationTokenSource, Uri } from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import * as path from 'path'
+import AntoraCompletionProvider from './antoraCompletionProvider'
+import { disposeAll } from '../../util/dispose'
+
+export class AntoraSupportManager {
+  private readonly _disposables: vscode.Disposable[] = []
+
+  public constructor (private readonly context: vscode.ExtensionContext) {
+    this.context = context
+    this._disposables.push(vscode.languages.registerCompletionItemProvider(
+      {
+        language: 'asciidoc',
+        scheme: 'file',
+      },
+      new AntoraCompletionProvider(),
+      '{'
+    ))
+    this._disposables.push(vscode.workspace.onDidOpenTextDocument(async (textDocument) => {
+      const workspaceConfiguration = vscode.workspace.getConfiguration('asciidoc', null)
+      // look for Antora settings in workspaceState
+      const workspaceState: vscode.Memento = this.context.workspaceState
+      const isEnableAntoraSupportSettingDefined = await workspaceState.get('antoraSupportSetting')
+      if (isEnableAntoraSupportSettingDefined === true) {
+        if (workspaceConfiguration.get('antora.enableAntoraSupport') === true) {
+          this.dispose()
+        } else {
+          // no op
+        }
+      } else {
+        if (await getValidConfig(textDocument)) {
+          const answer = await vscode.window.showInformationMessage('We detect that you are working with Antora. Do you want to active Antora support?', 'yes', 'no thanks')
+          if (answer !== undefined) {
+            await workspaceState.update('antoraSupportSetting', true)
+            const enableAntoraSupport = answer === 'yes' ? true : (answer === 'no thanks' ? false : undefined)
+            await workspaceConfiguration.update('antora.enableAntoraSupport', enableAntoraSupport)
+            if (enableAntoraSupport) {
+              this.dispose()
```

Comment:
Here we should register the completionItemProvider if it's not already registered. If the user does not want to enable Antora support then we should not register it.